tools/sanity_check/local_sanity_check.py.in

```diff
@@ -0,0 +1,269 @@
+SYS_NAME:   "{SYS_NAME}"
+BLK_PATHS:  "{BLK_PATHS}"
+ISSUE:      "{ISSUE}"
+SUGGEST:    "{SUGGEST}"
+"""
+
+    def __init__(self, blk_paths, vol, sys, msg, suggest):
+        self.blk_paths = blk_paths
+        self.vol = vol
+        self.sys = sys
+        self.msg = msg
+        self.suggest = suggest
+
+    def gen_fail_msg(self):
+        return Failure.OUTPUT_FORMAT.format(
+            **{
+                "VOL_ID": self.vol.id,
+                "VOL_NAME": self.vol.name,
+                "SYS_ID": self.sys.id,
+                "SYS_NAME": self.sys.name,
+                "BLK_PATHS": " ".join(self.blk_paths),
+                "ISSUE": self.msg,
+                "SUGGEST": self.suggest,
+            })
+
+
+class SanityCheck(unittest.TestCase):
+    def setUp(self):
+        uri = None
+        if os.getenv('LSMCLI_URI') is not None:
+            uri = os.getenv('LSMCLI_URI')
+        password = os.getenv('LSMCLI_PASSWORD')
+
+        if uri is None:
+            uri = "local://"
+            password = None
+
+        self.c = lsm.Client(uri, password)
+        self.syss = self.c.systems()
+        sys_hash = {}
+        cap_hash = {}
+        self.local_vol_hash = {}
+        for sys in self.syss:
+            sys_hash[sys.id] = sys
+            cap_hash[sys.id] = self.c.capabilities(sys)
+        for vol in self.c.volumes():
+            blk_paths = LocalDisk.vpd83_search(vol.vpd83)
+            if blk_paths:
+                self.local_vol_hash[vol.id] = {
+                    "blk_paths": blk_paths,
+                    "vol": vol,
+                    "sys": sys_hash[vol.system_id],
+                    "cap": cap_hash[vol.system_id],
+                }
+                print("\nDisk path of volume '%s(%s)': %s" %
+                      (vol.name, vol.id, " ".join(blk_paths)))
+        if not self.local_vol_hash:
+            self.skipTest("No local disk is managed by libstoragemgmt")
+
+    def tearDown(self):
+        self.c.close()
+
+    def _skip_current_test(self, messsage):
+        """
+        If skipTest is supported(new in python 2.7), skip this test with
+        provided message.
+        Silently return if not supported.
+        """
+        if hasattr(unittest.TestCase, 'skipTest') is True:
+            self.skipTest(messsage)
+        return
+
+    def _check_fail(self, fails):
+        spliter = '-' * 72 + "\n"
+        output = spliter.join(list(f.gen_fail_msg() for f in fails))
+        self.assertTrue(len(fails) == 0, "\n%s%s" % (spliter, output))
+
+    def test_volume_cache(self):
+        fails = []
+        flag_has_pass = False
+
+        for vol_id in self.local_vol_hash.keys():
+            flag_pass = True
+            sys = self.local_vol_hash[vol_id]["sys"]
+            cap = self.local_vol_hash[vol_id]["cap"]
+            vol = self.local_vol_hash[vol_id]["vol"]
+            blk_paths = self.local_vol_hash[vol_id]["blk_paths"]
+            if not supported(cap, [Cap.VOLUMES, Cap.VOLUME_CACHE_INFO]):
```

Comment:
I'm confused about this check.  In `setUp` we retrieve all the volumes without checking capability and even though `VOLUME_CACHE_INFO` is not supported a call is made to  `self.c.volume_cache_info()` on line 135?

Reply:
I missed a `continue`